cert.go

```diff
@@ -102,6 +102,7 @@ func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *t
 			return newConfig, nil
 		}
 
+		apisMu.RLock()
```

Comment:
Note that it will be called for each incoming connection, and I would rather avoid it. 

Can we lock only writers here, and keep readers not touched?

Reply:
Well, this is what happens if you use a global. All the other api operations obtain a read lock. If we only use the lock for writes, it's absolutely useless.

Reply:
@buger @mvdan 
I wander if we can benefit from using `sync.atomic`'s `Value` and `copy-on-write` approach showed here: https://golang.org/pkg/sync/atomic/#example_Value_readMostly ?

Also, kind of related to that comment - I was considering that approach https://golang.org/pkg/sync/atomic/#example_Value_config  for re-loading api definitions and use them to serve requests

Reply:
@dencoded make sense. But in fact, we modify apiSpec records only inside `apiLoader`, so we modify them all at once, and can replace in Value.